test/aec_unwrapped_test.py

```diff
@@ -0,0 +1,157 @@
+    return homogenizable
+
+
+def action_homogenizable(env, agents):
+    homogenizable = True
+    for agent in agents:
+        homogenizable = homogenizable and (
+            isinstance(env.action_space(agent), spaces.Box)
+            or isinstance(env.action_space(agent), spaces.Discrete)
+        )
+    return homogenizable
+
+
+def image_observation(env, agents):
+    imagable = True
+    for agent in agents:
+        if isinstance(env.observation_space(agent), spaces.Box):
+            imagable = imagable and (env.observation_space(agent).low.shape == 3)
+            imagable = imagable and (len(env.observation_space(agent).shape[2]) == 3)
+            imagable = imagable and (env.observation_space(agent).low == 0).all()
+            imagable = imagable and (env.observation_space(agent).high == 255).all()
+        else:
+            return False
+    return imagable
+
+
+def box_action(env, agents):
+    boxable = True
+    for agent in agents:
+        boxable = boxable and isinstance(env.action_space(agent), spaces.Box)
+    return boxable
+
+
+def not_dict_observation(env, agents):
+    is_dict = True
+    for agent in agents:
+        is_dict = is_dict and (isinstance(env.observation_space(agent), spaces.Dict))
+    return not is_dict
+
+
+def not_discrete_observation(env, agents):
+    is_discrete = True
+    for agent in agents:
+        is_discrete = is_discrete and (
+            isinstance(env.observation_space(agent), spaces.Discrete)
+        )
+    return not is_discrete
+
+
+def not_multibinary_observation(env, agents):
+    is_discrete = True
+    for agent in agents:
+        is_discrete = is_discrete and (
+            isinstance(env.observation_space(agent), spaces.MultiBinary)
+        )
+    return not is_discrete
+
+
+def unwrapped_check(env):
+    env.reset()
+    agents = env.agents
+    base_env = env.unwrapped
+
+    if image_observation(env, agents):
+        env = max_observation_v0(env, 2)
+        env = color_reduction_v0(env, mode="full")
+        env = normalize_obs_v0(env)
+
+    if box_action(env, agents):
+        env = clip_actions_v0(env)
+        env = scale_actions_v0(env, 0.5)
+
+    if observation_homogenizable(env, agents):
+        env = pad_observations_v0(env)
+        env = frame_stack_v1(env, 2)
+        env = agent_indicator_v0(env)
+        env = black_death_v3(env)
+
+    if (
+        not_dict_observation(env, agents)
+        and not_discrete_observation(env, agents)
+        and not_multibinary_observation(env, agents)
+    ):
+        env = dtype_v0(env, np.float16)
+        env = flatten_v0(env)
+        env = frame_skip_v0(env, 2)
+
+    if action_homogenizable(env, agents):
+        env = pad_action_space_v0(env)
+
+    env = clip_reward_v0(env, lower_bound=-1, upper_bound=1)
+    env = delay_observations_v0(env, 2)
+    env = sticky_actions_v0(env, 0.5)
+    env = nan_random_v0(env)
+    env = nan_zeros_v0(env)
+
+    assert env.unwrapped == base_env, f"Failed to unwrap {env}"
```

Comment:
Could you do something along the lines of `env.unwrapped.__class__ == DummyEnv`?

Reply:
Gotcha, will do.